contracts/test/integration/IntegrationTestPriceBoundPSMUSDC.t.sol

```diff
@@ -0,0 +1,333 @@
+        );
+        assertEq(
+            endingPSMUnderlyingBalance,
+            startingPSMUnderlyingBalance - amountOut
+        );
+    }
+
+    /// @notice redeem fails without approval
+    function testSwapVoltForUSDCFailsWithoutApproval() public {
+        vm.expectRevert(bytes("ERC20: transfer amount exceeds allowance"));
+
+        psm.redeem(address(this), mintAmount, mintAmount / 1e12);
+    }
+
+    /// @notice mint fails without approval
+    function testSwapUnderlyingForVoltFailsWithoutApproval() public {
+        vm.expectRevert(bytes("ERC20: transfer amount exceeds allowance"));
+
+        psm.mint(address(this), mintAmount, 0);
+    }
+
+    /// @notice withdraw erc20 fails without correct permissions
+    function testERC20WithdrawFailure() public {
+        vm.expectRevert(bytes("CoreRef: Caller is not a PCV controller"));
+
+        psm.withdrawERC20(address(underlyingToken), address(this), 100);
+    }
+
+    /// @notice withdraw erc20 succeeds with correct permissions
+    function testERC20WithdrawSuccess() public {
+        vm.prank(addresses.voltGovernorAddress);
+        core.grantPCVController(address(this));
+
+        uint256 startingBalance = underlyingToken.balanceOf(address(this));
+        psm.withdrawERC20(address(underlyingToken), address(this), mintAmount);
+        uint256 endingBalance = underlyingToken.balanceOf(address(this));
+
+        assertEq(endingBalance - startingBalance, mintAmount);
+    }
+
+    /// @notice set global rate limited minter fails when caller is governor and new address is 0
+    function testSetPCVDepositFailureZeroAddress() public {
+        vm.startPrank(addresses.voltGovernorAddress);
+
+        vm.expectRevert(
+            bytes("PegStabilityModule: Invalid new surplus target")
+        );
+        psm.setSurplusTarget(IPCVDeposit(address(0)));
+
+        vm.stopPrank();
+    }
+
+    /// @notice set PCV deposit fails when caller is governor and new address is 0
+    function testSetPCVDepositFailureNonGovernor() public {
+        vm.expectRevert(
+            bytes("CoreRef: Caller is not a governor or contract admin")
+        );
+        psm.setSurplusTarget(IPCVDeposit(address(0)));
+    }
+
+    /// @notice set PCV Deposit succeeds when caller is governor and underlying tokens match
+    function testSetPCVDepositSuccess() public {
+        vm.startPrank(addresses.voltGovernorAddress);
+
+        MockPCVDepositV2 newPCVDeposit = new MockPCVDepositV2(
+            address(core),
+            address(underlyingToken),
+            0,
+            0
+        );
+
+        psm.setSurplusTarget(IPCVDeposit(address(newPCVDeposit)));
+
+        vm.stopPrank();
+
+        assertEq(address(newPCVDeposit), address(psm.surplusTarget()));
+    }
+
+    /// @notice set mint fee succeeds
+    function testSetMintFeeSuccess() public {
+        vm.prank(addresses.voltGovernorAddress);
+        psm.setMintFee(100);
+
+        assertEq(psm.mintFeeBasisPoints(), 100);
```

Comment:
I had to check the previous mint fee basis points weren't 100, so use the existing and add one or something.
```suggestion
        uint256 newMintFee = psm.mintFeeBasisPoints() + 1;
        psm.setMintFee(newMintFee);

        assertEq(psm.mintFeeBasisPoints(), newMintFee);
```